src/msg/producer/writer/consumer_writer_test.go

```diff
@@ -87,7 +89,7 @@ func TestNewConsumerWriter(t *testing.T) {
 
 	w.Close()
 	// Make sure the connection is closed after closing the consumer writer.
-	_, err = w.conn.Read([]byte{})
+	_, err = w.writeState.conns[0].conn.Read([]byte{})
```

Comment:
Worth adding a test for multiple conns? Happy to `//TODO` it if it requires a bit of effort

Reply:
Will add todo.